pymc_bart/pgbart.py

```diff
@@ -456,17 +465,23 @@ def get_split_value(available_splitting_values, idx_data_points, missing_data):
 
 
 @njit
-def draw_leaf_value(y_mu_pred, m, norm, shape):
+def draw_leaf_value(y_mu_pred, x_mu, m, norm, shape, response):
     """Draw Gaussian distributed leaf values."""
+    linear_params = None
+    mu_mean = np.empty(shape)
     if y_mu_pred.size == 0:
-        return np.zeros(shape)
+        return np.zeros(shape), linear_params
 
     if y_mu_pred.size == 1:
         mu_mean = np.full(shape, y_mu_pred.item() / m)
     else:
-        mu_mean = fast_mean(y_mu_pred) / m
-
-    return norm + mu_mean
+        if response == "linear":
+            y_fit, linear_params = fast_linear_fit(x=x_mu, y=y_mu_pred)
+            mu_mean = fast_mean(y_fit.reshape(1, -1)) / m
```

Comment:
The model runs in for the linear response in the scalar case (see tests). However, it fails on the test with non-trivial shapes. I am actually not convinced with this piece of code to define the output. Do you have any suggestions?

Reply:
solved in https://github.com/pymc-devs/pymc-bart/pull/79/commits/f2c546a33059fd8f96375f5fb8a81b2f364a7fc0